src/main/java/ch/njol/skript/expressions/ExprNoDamageTicks.java

```diff
@@ -18,78 +18,72 @@
  */
 package ch.njol.skript.expressions;
 
-import org.bukkit.entity.LivingEntity;
-import org.bukkit.event.Event;
-import org.eclipse.jdt.annotation.Nullable;
-
 import ch.njol.skript.classes.Changer.ChangeMode;
 import ch.njol.skript.doc.Description;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.util.Timespan;
 import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
 
-@Name("No Damage Ticks")
-@Description("The number of ticks that an entity is invulnerable to damage for.")
+@Name("No Damage Duration")
+@Description("How much time an entity is invulnerable for.")
 @Examples({"on damage:",
-		"	set victim's invulnerability ticks to 20 #Victim will not take damage for the next second"})
+		"	set victim's invulnerability time to 1 second #Victim will not take damage for the next second"})
 @Since("2.5")
-public class ExprNoDamageTicks extends SimplePropertyExpression<LivingEntity, Long> {
+public class ExprNoDamageTicks extends SimplePropertyExpression<LivingEntity, Timespan> {
 	
 	static {
-		register(ExprNoDamageTicks.class, Long.class, "(invulnerability|no damage) tick[s]", "livingentities");
+		register(ExprNoDamageTicks.class, Timespan.class, "(invulnerability|no damage) (time|duration)", "livingentities");
 	}
 
 	@Override
-	public Long convert(LivingEntity e) {
-		return (long) e.getNoDamageTicks();
+	public Timespan convert(LivingEntity entity) {
+		return Timespan.fromTicks_i(Math.max(entity.getNoDamageTicks(), 0));
```

Comment:
It can be set to negative values ([here](https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/browse/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java#336)) and getting doesn't check for it either ([here](https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/browse/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java#331)).